perfkitbenchmarker/rackspace/rackspace_disk.py

```diff
@@ -0,0 +1,209 @@
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    delete_cmd = [FLAGS.nova_path, 'volume-delete', self.name]
+    vm_util.IssueCommand(delete_cmd, env=nova_env)
+
+  def _Exists(self):
+    """Returns true if the volume exists."""
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    getdisk_cmd = [FLAGS.nova_path, 'volume-show', self.name]
+    stdout, _, _ = vm_util.IssueCommand(getdisk_cmd, env=nova_env)
+    if stdout.strip() == '':
+      return False
+    volume = util.ParseNovaTable(stdout)
+    return 'status' in volume and volume['status'] == 'available'
+
+  def Attach(self, vm):
+    """Attaches the volume to a VM.
+
+    Args:
+      vm: The RackspaceVirtualMachine instance to which the volume will be
+          attached.
+    """
+    self.attached_vm_name = vm.name
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    attach_cmd = [
+        FLAGS.nova_path,
+        'volume-attach',
+        self.attached_vm_name,
+        self.id, ""]
+    vm_util.IssueRetryableCommand(attach_cmd, env=nova_env)
+
+    volume_show_cmd = [FLAGS.nova_path, 'volume-show', self.name]
+
+    attached = False
+    for i in xrange(12):
+      stdout, _, _ = vm_util.IssueCommand(volume_show_cmd, env=nova_env)
+      attrs = stdout.split('\n')
+      for attr in attrs[3:-2]:
+        pv = [v.strip() for v in attr.split('|')
+              if v != '|' and v != '']
+        if pv[0] == 'attachments' and vm.id in pv[1]:
+            attached = True
+      if attached:
+        return
+      time.sleep(10)
+
+    if self.num_retried_attach > 5:
+      raise Exception("Failed to attach all scratch disks to vms. "
+                      "Exiting...")
+    self.num_retried_attach += 1
+    self.Attach(vm)
+
+  def Detach(self):
+      """Detaches the disk from a VM."""
+      nova_env = os.environ.copy()
+      nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+      detach_cmd = [
+          FLAGS.nova_path,
+          'volume-detach',
+          self.attached_vm_name,
+          self.id]
+      vm_util.IssueRetryableCommand(detach_cmd, env=nova_env)
+      self.attached_vm_name = None
+
+  def GetDevicePath(self):
+    """Returns the path to the device inside the VM."""
+    nova_env = os.environ.copy()
+    nova_env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    getdisk_cmd = [FLAGS.nova_path, 'volume-show', self.name]
+    stdout, _, _ = vm_util.IssueCommand(getdisk_cmd, env=nova_env)
+    if stdout.strip() == '':
+      return False
+    volume = util.ParseNovaTable(stdout)
+    if 'attachments' in volume and volume['attachments']:
+      attachment_list = json.loads(volume['attachments'])
+      if attachment_list:
+        for attachment in attachment_list:
+          if 'volume_id' in attachment and attachment['volume_id'] == self.id:
+            return attachment['device']
+
+    return ''
```

Comment:
This looks like it could lead to odd errors if it ends up at the default return unexpectedly. Consider raising an exception if this should never happen, or None if this is mainly for __str__ convenience?